model_analyzer/config/run/model_run_config.py

```diff
@@ -115,6 +115,17 @@ def is_legal_combination(self):
             logger.debug(
                 f"Illegal model run config because client batch size {perf_batch_size} is greater than model max batch size {max_batch_size}"
             )
+            return legal
```

Comment:
I'd like this refactored into two separate methods: check_for_client_vs_model_batch_size & check_for_preferred_vs_model_batch_size

Each method should return legal as a bool.

Reply:
Created two separate methods check_for_client_vs_model_batch_size & check_for_preferred_vs_model_batch_size